### PR TITLE
docs: more comprehensive description of username normalization

### PR DESCRIPTION
**Pull request by unknwon**
Follow up of #5771.

### CODE REVIEW

File: cmd/frontend/auth/auth.go [hunk 1]
```diff
@@ -61,8 +61,9 @@ func composeMiddleware(middlewares ...*Middleware) *Middleware {
 // username formatting rules (based on, but not identical to
 // https://help.github.com/enterprise/2.11/admin/guides/user-management/using-ldap/#username-considerations-with-ldap):
 //
-// - If there is exactly one `@` in the username, any portion since the `@` character is removed
 // - Any characters not in `[a-zA-Z0-9-.]` are replaced with `-`
+// - Usernames with exactly one `@` character, any portion since the `@` character is removed
```
Comment (christinaforney) [inline cmd/frontend/auth/auth.go]:
```suggestion
// - Usernames with exactly one `@` character are interpreted as an email address, so the username will be extracted by truncating at the `@` character.
```

File: doc/admin/auth/index.md [hunk 2]
```diff
@@ -314,8 +314,10 @@ Some proxies add a prefix to the username header value. For example, Google IAP
 
 Usernames on Sourcegraph are normalized according to the following rules.
 
-- If there is exactly one `@` in the username, any portion since the `@` character is removed
+
 - Any characters not in `[a-zA-Z0-9-.]` are replaced with `-`
+- Usernames with exactly one `@` character, any portion since the `@` character is removed
```
Comment (christinaforney) [inline doc/admin/auth/index.md]:
```suggestion
- Usernames with exactly one `@` character are interpreted as an email address, so the username will be extracted by truncating at the `@` character.
```

File: cmd/frontend/auth/auth.go [hunk 3]
```diff
@@ -61,8 +61,9 @@ func composeMiddleware(middlewares ...*Middleware) *Middleware {
...
+// - Usernames with two or more `@` characters, they are treated as non-standard characters and both are replaced with `-`
```
Comment (christinaforney) [inline cmd/frontend/auth/auth.go]:
```suggestion
// - Usernames with two or more `@` characters are not considered an email address, so the `@` will be treated as a non-standard character and be replaced with `-`
```

File: doc/admin/auth/index.md [hunk 4]
```diff
@@ -314,8 +314,10 @@ Some proxies add a prefix to the username header value. For example, Google IAP
...
+- Usernames with two or more `@` characters, they are treated as non-standard characters and both are replaced with `-`
```
Comment (christinaforney) [inline doc/admin/auth/index.md]:
```suggestion
- Usernames with two or more `@` characters are not considered an email address, so the `@` will be treated as a non-standard character and be replaced with `-`
```

**Comment by unknwon** [comment]
Thanks for the suggestions :)

**Pull request merged**